### PR TITLE
Add examples for fakes

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Will update when `fake` goes GA.

### CODE REVIEW

**Review by chlowell — APPROVED** [review]
This is really helpful 🥇. I suppose for visibility we'll link to it in readmes and azcore/fake?

File: sdk/samples/fakes/README.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Examples using fakes
+
+The examples in this module demonstrate using the `fake` subpackage for creating unit tests.
+While the examples use `armcompute`, the patterns are applicable to any SDK with a `fake` subpackage.
```
[the remark this answers is not in the document]
Comment (jhendrixMSFT) [reply sdk/samples/fakes/README.md]: Fixed.

**Comment by jhendrixMSFT** [comment]
Yes, will add links from `azcore` and the `README.md` files for other SDK modules (e.g. [here](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/resourcemanager/compute/armcompute/README.md#fakes)).

**Pull request merged**